### ISSUE TITLE
Homebrew installation fails

### ISSUE THREAD

**Issue by melgu** [simonw/llm#101]
The installation fails when trying via Homebrew.

<details>
<summary>Terminal output</summary>

```
>>> brew install simonw/llm/llm

Found existing alias for "brew install". You should use: "brewi"
==> Tapping simonw/llm
Cloning into '/opt/homebrew/Library/Taps/simonw/homebrew-llm'...
remote: Enumerating objects: 146, done.
remote: Counting objects: 100% (146/146), done.
remote: Compressing objects: 100% (109/109), done.
remote: Total 146 (delta 45), reused 47 (delta 12), pack-reused 0
Receiving objects: 100% (146/146), 30.44 KiB | 1.90 MiB/s, done.
Resolving deltas: 100% (45/45), done.
Tapped 4 formulae (21 files, 60.5KB).
==> Downloading https://formulae.brew.sh/api/formula.jws.json
############################################################################################# 100.0%
==> Fetching simonw/llm/llm
==> Downloading https://files.pythonhosted.org/packages/c2/fd/1ff4da09ca29d8933fda3f3514980357e25419
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/ae/67/0952ed97a9793b4958e5736f6d2b346b414a2c
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/42/97/41ccb6acac36fdd13592a686a21b311418f786
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/54/6e/9678f7b2993537452710ffb1750c62d2c26df4
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/97/90/81f95d5f705be17872843536b1868f351805ac
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/93/71/752f7a4dd4c20d6b12341ed1732368546bc0ca
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/2a/53/cf0a48de1bdcf6ff6e1c9a023f5f523dfe303e
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/77/88/b0cc5fe95c31c301e9823ea9b028f669c0dcfa
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/3d/da/f3bbf30f7e71d881585d598f67f4424b2cc4c6
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/e9/10/d629476346112b85c912527b9080944fd2c39a
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/8b/e1/43beb3d38dba6cb420cefa297822eac205a277
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/4a/15/bd620f7a6eb9aa5112c4ef93e7031bcd071e06
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/89/ee/84bbd0161090f0f24e8a2ac175e6b6a936289e
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/8a/42/8f2833655a29c4e9cb52ee8a2be04ceac61bcf
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/3c/39/e526f3ca06e062e0ad773555f5349ff5d9ed50
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/66/68/9703e44f0bcc29eeaacb1c063675687524646a
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/4c/c4/13b4776ea2d76c115c1d1b84579f3764ee6d57
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/e8/8b/0580d8ee0a73a3f3869488856737c429cbaa08
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/36/2b/61d51a2c4f25ef062ae3f74576b01638bebad5
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/9d/be/10918a2eac4ae9f02f6cfe6414b7a155ccd8f7
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/c2/6d/9dad6c3b433ab8912ace969c66abd595f8e0a2
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/99/48/b974a270fd710e62c5ab863dc216cf3ffe26b5
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/ec/fe/802052aecb21e3797b8f7902564ab6ea0d60ff
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/3d/78/81191f56abb7d3d56963337dbdff6aa4f55805
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/3c/8b/0111dd7d6c1478bf83baa1cab85c686426c7a6
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/d6/af/3b4cfedd46b3addab52e84a71ab26518272c23
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/5f/3f/04b3c5e57844fb9c034b09c5cb6d2b43de5d64
############################################################################################# 100.0%
==> Downloading https://files.pythonhosted.org/packages/27/0f/f4afb3bc87c17b93d913e56ea09dedbad06a4f
############################################################################################# 100.0%
==> Installing llm from simonw/llm
==> python3 -m venv --system-site-packages /opt/homebrew/Cellar/llm/0.5/libexec
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
==> /opt/homebrew/Cellar/llm/0.5/libexec/bin/pip install -v --no-deps --no-binary :all: --use-featur
Last 15 lines from /Users/melvin/Library/Logs/Homebrew/llm/17.pip:

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /opt/homebrew/Cellar/llm/0.5/libexec/bin/python3.11 /opt/homebrew/Cellar/llm/0.5/libexec/lib/python3.11/site-packages/pip/__pip-runner__.py install --ignore-installed --no-user --prefix /private/tmp/pip-build-env-323dud3w/overlay --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- 'maturin>=1,<2' 'typing-extensions >=4.6.0,!=4.7.0'
  cwd: [inherit]
  Installing build dependencies: finished with status 'error'
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.0.1 -> 23.1.2
[notice] To update, run: python3.11 -m pip install --upgrade pip

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/simonw/homebrew-llm/issues
```

</details>

**Comment by simonw** [comment]
Oh dear... I think that might be caused by the new `pydantic` 2.0 dependency, which uses Rust.

**Comment by simonw** [comment]
Trying this workaround: https://til.simonwillison.net/homebrew/auto-formulas-github-actions#user-content-a-formula-with-rust-compiled-dependencies

**Comment by simonw** [comment]
The downside of my current Homebrew installation flow is that it doesn't use Python wheels, which actually makes it massively slower than installing using `pip` or `pipx`.

**Comment by simonw** [comment]
That change fixed it - it's really slow though, see #102.

**Comment by melgu** [comment]
Can confirm that it works now. Can also confirm, that it's slow 😄
Thanks for the very quick response and fix 😃
